store/pg_engine_test.go

```diff
@@ -16,47 +17,101 @@ import (
 
 	tests := []struct {
-		versions                []semver.Version
+		names                   []string
 		releaseCutSchemaVersion semver.Version
 		currentVersion          semver.Version
 		want                    []semver.Version
 	}{
 		{
-			versions,
-			semver.MustParse("1.0.0"),
-			semver.MustParse("1.0.0"),
-			nil,
+			names:                   names,
+			releaseCutSchemaVersion: semver.MustParse("1.0.0"),
+			currentVersion:          semver.MustParse("1.0.0"),
+			want:                    []semver.Version{semver.MustParse("1.0.0")},
 		},
 		{
-			versions,
-			semver.MustParse("1.1.1"),
-			semver.MustParse("1.1.1"),
-			nil,
+			names:                   names,
+			releaseCutSchemaVersion: semver.MustParse("1.3.3"),
+			currentVersion:          semver.MustParse("1.3.0"),
+			want:                    []semver.Version{semver.MustParse("1.3.0")},
 		},
 		{
-			versions,
-			semver.MustParse("1.1.1"),
-			semver.MustParse("1.0.0"),
-			[]semver.Version{semver.MustParse("1.1.0"), semver.MustParse("1.1.1")},
+			names:                   names,
+			releaseCutSchemaVersion: semver.MustParse("1.3.0"),
+			currentVersion:          semver.MustParse("1.0.3"),
+			want:                    []semver.Version{semver.MustParse("1.0.0"), semver.MustParse("1.1.0"), semver.MustParse("1.2.0"), semver.MustParse("1.3.0")},
 		},
 		{
-			versions,
-			semver.MustParse("1.3.0"),
-			semver.MustParse("1.0.0"),
-			[]semver.Version{semver.MustParse("1.1.0"), semver.MustParse("1.1.1"), semver.MustParse("1.2.0"), semver.MustParse("1.3.0")},
+			names:                   names,
+			releaseCutSchemaVersion: semver.MustParse("1.3.0"),
+			currentVersion:          semver.MustParse("1.2.2"),
+			want:                    []semver.Version{semver.MustParse("1.2.0"), semver.MustParse("1.3.0")},
 		},
 	}
 
 	for _, test := range tests {
-		migrateVersions, _ := getMigrationVersions(test.versions, test.releaseCutSchemaVersion, test.currentVersion)
+		migrateVersions, _, _ := getMinorMigrationVersions(test.names, test.releaseCutSchemaVersion, test.currentVersion)
 		require.Equal(t, test.want, migrateVersions)
 	}
 }
 
+func TestGetMiorVersions(t *testing.T) {
+	tests := []struct {
+		names []string
+		want  []semver.Version
+	}{
+		{
+			names: []string{fmt.Sprintf("migration/dev/%s", latestDataFile), fmt.Sprintf("migration/dev/%s", latestSchemaFile), "migration/dev/1.1.2", "migration/dev/1.0.0"},
+			want:  []semver.Version{semver.MustParse("1.0.0"), semver.MustParse("1.1.2")},
+		},
+		{
+			names: []string{fmt.Sprintf("migration/release/%s", latestDataFile), fmt.Sprintf("migration/dev/%s", latestSchemaFile)},
+			want:  nil,
+		},
+	}
+
+	for _, test := range tests {
+		got, _ := getMiorVersions(test.names)
+		require.Equal(t, test.want, got)
+	}
+}
+
+func TestGetPatchVersions(t *testing.T) {
+	tests := []struct {
+		names          []string
+		minorVersion   semver.Version
+		currentVersion semver.Version
+		want           []patchVersion
+	}{
+		{
```

Comment:
add an invalid file format?

Reply:
Also add a test case where minorVersion and currentVersion only differs in the PATCH version.